pkg/defaults/init.go

```diff
@@ -22,7 +23,29 @@ func InitIntegration(portClient *cli.PortClient, applicationConfig *port.Config)
 	if err != nil {
 		return err
 	}
-	
+
+	// Verify Port origin is supported via feature flags
+	if applicationConfig.CreatePortResourcesOrigin == port.CreatePortResourcesOriginPort {
+		klog.Info("Resources origin is set to be Port, verifying integration is supported")
+		featureFlags, err := org_details.GetOrganizationFeatureFlags(portClient)
+		if err != nil {
+			return err
+		}
+
+		hasProvisionedDefaultsFlag := false
+		for _, flag := range featureFlags {
+			if flag == port.OrgUseProvisionedDefaultsFeatureFlag {
+				hasProvisionedDefaultsFlag = true
+				break
+			}
+		}
+
+		if !hasProvisionedDefaultsFlag {
+			klog.Info("Port origin for Integration is not supported, changing resources origin to use K8S")
+			applicationConfig.CreatePortResourcesOrigin = port.CreatePortResourcesOriginK8S
+		}
+	}
```

Comment:
Can you put that in a different function, it could be something generic like `isFeatureFalgSupported`